cloud/linode/instances.go

```diff
@@ -29,7 +75,33 @@ func (e instanceNoIPAddressesError) Error() string {
 	return fmt.Sprintf("instance %d has no IP addresses", e.id)
 }
 
+func (i *instances) linodeByName(nodeName types.NodeName) (*linodego.Instance, error) {
+	i.nodeCache.RLock()
+	defer i.nodeCache.RUnlock()
+	for _, node := range i.nodeCache.nodes {
+		if node.Label == string(nodeName) {
+			return node, nil
+		}
+	}
+
+	return nil, cloudprovider.InstanceNotFound
```

Comment:
I wonder if now we should still handle this as `cloudprovider.InstanceNotFound` since technically it's not found from our own cache, and not the provider.. It may be all the same either way.

Reply:
I see the cache as an implementation detail - the fact that we're not interrogating the API live is just our internal logic.

What would you suggest as an alternative? `errors.New("instance not found")`, which would suggest we didn't find it ourselves, not that it's not present upstream within the cloud provider?